tests/test_layers.py

```diff
@@ -141,23 +142,22 @@ def test_projected_adaptive_softmax(vocab_size, cutoffs, embed_size, in_units, d
                                             cutoffs=cutoffs,
                                             embed_size=embed_size,
                                             in_units=in_units,
-                                            div_val=div_val,
-                                            params=embed_layer.collect_params('.*_inter_proj'))
+                                            div_val=div_val)
+    layer_with_shared_proj.share_parameters(embed_layer.collect_params('.*_inter_proj'))
     layer_with_shared_embed = \
         ProjectedAdaptiveLogSoftmaxWithLoss(vocab_size=vocab_size,
                                             cutoffs=cutoffs,
                                             embed_size=embed_size,
                                             in_units=in_units,
-                                            div_val=div_val,
-                                            params=embed_layer.collect_params('.*_embed'))
+                                            div_val=div_val)
+    layer_with_shared_embed.share_parameters(embed_layer.collect_params('.*_embed'))
     layer_with_shared_proj_embed = \
         ProjectedAdaptiveLogSoftmaxWithLoss(vocab_size=vocab_size,
                                             cutoffs=cutoffs,
                                             embed_size=embed_size,
                                             in_units=in_units,
-                                            div_val=div_val,
-                                            params=embed_layer.collect_params(
-                                                '(.*_embed|.*_inter_proj)'))
+                                            div_val=div_val)
+    layer_with_shared_proj_embed.share_parameters(embed_layer.collect_params('(.*_embed|.*_inter_proj)'))
```

Comment:
It seems that we cant locate to these parameters since prefix are removed 
https://github.com/leezu/gluon-nlp/blob/a79101da3a40d5212e419fa1f46a40e9ad3e7eb3/src/gluonnlp/layers.py#L916-L917

Reply:
It's because the regex needs to be updated. It contains  `_` character which is no longer used and should be replaced by `\.`.